### PR TITLE
Fix invoice steuernews

### PR DESCRIPTION
**Pull request by niner**
This fixes an easily reproducible issue with t/invoice_steuernews.t which is caused by an unfinished previous page load and improvements I discovered along the way.
However, while I think it's the right direction, it causes regressions in other tests that need fixing before it can be merged.

### CODE REVIEW

File: lib/WWW/WebKit2/Locator.pm [hunk 1]
```diff
@@ -338,11 +338,11 @@ sub fire_event {
         }, { once: true });
         var event = new Event("' . $event_type . '", { "bubbles": true, "cancelable": true });
         element.dispatchEvent(event);
-    ';
+        ';
```
Comment (jscarty) [inline lib/WWW/WebKit2/Locator.pm:341]: Is the white space needed or for lining up with the rest of the JS?

File: lib/WWW/WebKit2.pm [hunk 2]
```diff
@@ -343,6 +348,22 @@ sub init_webkit {
...
 
+    $self->view->signal_connect('decide-policy' => sub {
+        my ($view, $decision, $type) = @_;
+        if ($type eq 'navigation-action') {
+            my $action = $decision->get_navigation_action;
+            die "Already running a navigation action to " .
+                $self->active_navigation_action
+                . " when requested "
+                . $action->get_navigation_type . ' ' . $action->get_request->get_uri
+                if $self->active_navigation_action and not $self->navigation_action->is_redirect;
```
Comment (jscarty) [inline lib/WWW/WebKit2.pm]: Is `$self->navigation_action` a typo should it be `$action->navigation_action`?

`$self->navigation_action` is never set.

**Review by jscarty — COMMENTED** [review]
I am taking a look at the regressions.

**Review by RedGreatApe — APPROVED** [review]
(Y)

**Pull request merged**